pkg/controller/data_download_controller.go

```diff
@@ -416,6 +422,43 @@ func (r *DataDownloadReconciler) findSnapshotRestoreForPod(podObj client.Object)
 	return requests
 }
 
+func (r *DataDownloadReconciler) FindDataDownloads(ctx context.Context, client client.Client) ([]*velerov2alpha1api.DataDownload, error) {
+	pods := &v1.PodList{}
+	var dataDownloads []*velerov2alpha1api.DataDownload
+	if err := client.List(ctx, pods); err != nil {
```

Comment:
Below way performs better:
List pods by filtering label with `velerov1api.DataDownloadLabel`
Then we get a much small collection of pods to go with `findDataDownloadByPod`

Reply:
For controller runtime client, there is no selector that could filter one resource by one fixed key and unfixed value

Reply:
I think the list method has a ListOption parameter which can be used to specify labels? Just like this one https://github.com/vmware-tanzu/velero/blob/7deae4cbf5443fabe65eda165ac41594598ec0d5/pkg/controller/backup_controller.go#L161C3-L161C3

Reply:
ok, modified